sdk/storage/azure-storage-blob/src/test/java/com/azure/storage/blob/specialized/BlobBaseAPITest.groovy

```diff
@@ -0,0 +1,667 @@
+            .setFieldQuote(fieldQuote as char)
+            .setHeadersPresent(headersPresent)
+        uploadCsv(ser, 32)
+        def expression = "SELECT * from BlobStorage"
+
+        ByteArrayOutputStream downloadData = new ByteArrayOutputStream()
+        bc.download(downloadData)
+        byte[] downloadedData = downloadData.toByteArray()
+
+        /* Input Stream. */
+        when:
+        InputStream qqStream = bc.openQueryInputStream(expression, new BlobQueryOptions().setInputSerialization(ser).setOutputSerialization(ser))
+        byte[] queryData = readFromInputStream(qqStream, downloadedData.length)
+
+        then:
+        notThrown(IOException)
+        if (headersPresent) {
+            /* Account for 16 bytes of header. */
+            for (int j = 16; j < downloadedData.length; j++) {
+                assert queryData[j - 16] == downloadedData[j]
+            }
+            for (int k = downloadedData.length - 16; k < downloadedData.length; k++) {
+                assert queryData[k] == 0
+            }
+        } else {
+            queryData == downloadedData
+        }
+
+        /* Output Stream. */
+        when:
+        OutputStream os = new ByteArrayOutputStream()
+        bc.queryWithResponse(os, expression, new BlobQueryOptions().setInputSerialization(ser).setOutputSerialization(ser), null, null)
+        byte[] osData = os.toByteArray()
+
+        then:
+        notThrown(BlobStorageException)
+        if (headersPresent) {
+            assert osData.length == downloadedData.length - 16
+            /* Account for 16 bytes of header. */
+            for (int j = 16; j < downloadedData.length; j++) {
+                assert osData[j - 16] == downloadedData[j]
+            }
+        } else {
+            osData == downloadedData
+        }
+
+        // To calculate the size of data being tested = numCopies * 32 bytes
+        where:
+        recordSeparator | columnSeparator | escapeChar | fieldQuote | headersPresent || _
+        '\n'            | ','             | '\0'       | '\0'       | false          || _ /* Default. */
+        '\n'            | ','             | '\0'       | '\0'       | true           || _ /* Headers. */
+        '\t'            | ','             | '\0'       | '\0'       | false          || _ /* Record Separators. */
+        '\r'            | ','             | '\0'       | '\0'       | false          || _
+        '<'             | ','             | '\0'       | '\0'       | false          || _
+        '>'             | ','             | '\0'       | '\0'       | false          || _
+        '&'             | ','             | '\0'       | '\0'       | false          || _
+        '\\'            | ','             | '\0'       | '\0'       | false          || _
+        ','             | '.'             | '\0'       | '\0'       | false          || _ /* Column separator. */
+//        | ','             | '\0'       | '\\'       | false          || _ /* Field quote. */
```

Comment:
i'm not sure why or how some of my other cases got deleted. I'll add them back. Also forgot to add to file